pkg/util/chunk/column.go

```diff
@@ -245,6 +248,54 @@ func (c *Column) appendNullBitmap(notNull bool) {
 	}
 }
 
+// Reserve allocates some memory for the column
+func (c *Column) Reserve(nullBitmapExpectedCap int64, dataExpectedCap int64, offsetExpectedCap int64) {
+	currentNullBitmapLen := int64(len(c.nullBitmap))
+	nullBitmapExpectedCap += currentNullBitmapLen
+	if int64(cap(c.nullBitmap)) < nullBitmapExpectedCap {
+		c.nullBitmap = slices.Grow(c.nullBitmap, int(nullBitmapExpectedCap))
+	}
+
+	currentDataLen := int64(len(c.data))
+	dataExpectedCap += currentDataLen
+	if int64(cap(c.data)) < dataExpectedCap {
+		c.data = slices.Grow(c.data, int(dataExpectedCap))
+	}
+
+	currentOffsetLen := int64(len(c.offsets))
+	offsetExpectedCap += currentOffsetLen
+	if int64(cap(c.offsets)) < offsetExpectedCap {
+		c.offsets = slices.Grow(c.offsets, int(offsetExpectedCap))
+	}
+}
+
+func (c *Column) calculateLenDeltaForAppendCellNTimesForNullBitMap(times int) int64 {
+	nullBitMapLenDelta := int64(0)
+	if times == 1 {
```

Comment:
if `calculateLenDeltaForAppendCellNTimesForNullBitMap` is calculated once for a chunk, i think we don't need to check `times == 1` here

Reply:
> if `calculateLenDeltaForAppendCellNTimesForNullBitMap` is calculated once for a chunk, i think we don't need to check `times == 1` here

done